mitodl/mitxonline
# 872: checkout remove clear discount and a few other tweaks

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/872
https://github.com/mitodl/mitxonline/issues/871

#### What's this PR do?

- Removes the "Clear Discount" option from the Cart page.
- Removes the `clear_discount` API endpoint.
- Updates the Cart Item card to display additional fields (see https://github.com/mitodl/mitxonline/issues/871)

#### How should this be manually tested?

- Login as a user
- Add a course run to your Cart
- View your Cart and ensure that you no longer see the option to "Clear Discount"
- Add a discount, then ensure that you still no longer see the option to "Clear Discount"
- Attempt to apply another discount which is of lesser value than the current discount.  Ensure that the lesser value discount does not apply.
- Ensure that you see the UI additions specified in https://github.com/mitodl/mitxonline/issues/871

![Screen Shot 2022-08-19 at 8 36 11](https://user-images.githubusercontent.com/8311573/185619610-bfdd525f-0429-41ed-9828-15541997c143.png)
